dask_cuda/tests/test_spill.py

```diff
@@ -264,26 +289,35 @@ async def test_cudf_cluster_device_spill(params):
                 await wait(cdf2)
 
                 del cdf
+                gc.collect()
 
-                host_chunks = await client.run(
-                    lambda dask_worker: len(dask_worker.data.host)
+                await client.run(
+                    assert_host_chunks,
+                    params["spills_to_disk"],
                 )
-                disk_chunks = await client.run(
-                    lambda dask_worker: len(dask_worker.data.disk or list())
+                await client.run(
+                    assert_disk_chunks,
+                    params["spills_to_disk"],
                 )
-                for hc, dc in zip(host_chunks.values(), disk_chunks.values()):
-                    if params["spills_to_disk"]:
-                        assert dc > 0
-                    else:
-                        assert hc > 0
-                        assert dc == 0
 
                 await client.run(
-                    lambda dask_worker: worker_assert(dask_worker, nbytes, 32, 2048)
+                    worker_assert,
+                    nbytes,
+                    32,
+                    2048,
                 )
 
                 del cdf2
 
-                await client.run(
-                    lambda dask_worker: delayed_worker_assert(dask_worker, 0, 0, 0)
-                )
+                while True:
+                    try:
+                        await client.run(
+                            delayed_worker_assert,
+                            0,
+                            0,
+                            0,
+                        )
+                    except AssertionError:
+                        gc.collect()
+                    else:
+                        break
```

Comment:
Do we want a max number of tries?

```suggestion
                for _ in range(10):
                    try:
                        await client.run(
                            delayed_worker_assert,
                            0,
                            0,
                            0,
                        )
                    except AssertionError:
                        gc.collect()
                    else:
                        return
```

Reply:
I thought about this, but then I decided in favor of a timeout, as per description: "thus invoke garbage collection after del until it memory is actually released or the test times out.".

I think timing out is overall more appropriate than an arbitrarily-defined count. What do you think?

Reply:
The timeout is defined by the `gen_test` decorator, and that reminded me we can [now set it back to the 30 seconds it was before](https://github.com/rapidsai/dask-cuda/pull/1255/commits/7a12c4338686ac08cdad97fd882491729db6309c).

Reply:
Sounds good